xds/internal/httpfilter/httpfilter.go

```diff
@@ -50,6 +50,11 @@ type Filter interface {
 	// not accept a custom type.  The resulting FilterConfig will later be
 	// passed to Build.
 	ParseFilterConfigOverride(proto.Message) (FilterConfig, error)
+	// IsTerminal returns whether this Filter is terminal or not (i.e. it must
+	// be last filter in the filter chain). Currently, grpc only supports the
+	// router filter as a terminal filter, as per A39, although this may change
```

Comment:
Does this comment saying "router filter is the only terminal filter" need to be part of the interface definition?

Also, isnt HCM the terminal filter on the server?